services/i18n.js

```diff
@@ -1,57 +1,91 @@
-const has = require('lodash/has');
-const get = require('lodash/get');
-
+const fs = require('fs');
+const path = require('path');
+const debug = require('debug')('talk:services:i18n');
+const accepts = require('accepts');
+const _ = require('lodash');
 const yaml = require('yamljs');
+const plugins = require('./plugins');
+const {DEFAULT_LANG} = require('../config');
 
-const da = yaml.load('./locales/da.yml');
-const es = yaml.load('./locales/es.yml');
-const en = yaml.load('./locales/en.yml');
-const fr = yaml.load('./locales/fr.yml');
-const pt_BR = yaml.load('./locales/pt_BR.yml');
+const resolve = (...paths) => path.resolve(path.join(__dirname, '..', 'locales', ...paths));
 
-const accepts = require('accepts');
+// Load all the translations.
+let translations = fs.readdirSync(resolve())
+
+  // Resolve all the filenames relative the the locales directory.
+  .map((filename) => resolve(filename))
+
+  // Translations are only yml/yaml files.
+  .filter((filename) => /\.(yaml|yml)$/.test(filename))
+
+  // Load the translation files from disk.
+  .map((filename) => fs.readFileSync(filename, 'utf8'))
+
+  // Load the translation files.
+  .reduce((packs, contents) => {
 
-// default language
-let defaultLanguage = 'en';
-let language = defaultLanguage;
-const languages = ['en', 'da', 'es', 'fr', 'pt_BR'];
+    const pack = yaml.parse(contents);
 
-const translations = Object.assign(en, es, fr, pt_BR, da);
+    return _.merge(packs, pack);
+  }, {});
+
+// Create a list of all supported translations.
+const languages = Object.keys(translations);
+
+let loadedPluginTranslations = false;
+const loadPluginTranslations = () => {
+  if (loadedPluginTranslations) {
+    return;
+  }
+
+  // Load the plugin translations.
+  plugins.get('server', 'translations').forEach(({plugin, translations: filename}) => {
+    debug(`added plugin '${plugin.name}'`);
+
+    const pack = yaml.parse(fs.readFileSync(filename, 'utf8'));
```

Comment:
I think you need to deal with resolving the path relative to the plugin here.

Reply:
It's the expectation that the plugin will use something like ``` path.join(__dirname, 'client/translations.yml')``` to resolve it in file.